syntax = "proto3";

package s1see;

import "signal_message.proto";

message SpoolRecord {
    int32 partition = 1;       // Partition number
    int64 offset = 2;          // Monotonic offset within partition
    int64 ts_append = 3;       // Timestamp when appended to spool
    SignalMessage message = 4; // The signal message
}

message SpoolOffset {
    int32 partition = 1;
    int64 offset = 2;
    int64 frame_number = 3;  // Frame/packet number from source (e.g., PCAP frame number)
}

message EvidenceChain {
    repeated SpoolOffset offsets = 1; // All spool offsets that contributed to this event
}

